### ISSUE TITLE
Backing up multiple snapper configurations with different retention settings to the same borg repo is buggy

### ISSUE THREAD

**Issue by rkunze** [enzingerm/snapborg#37]
With snapborg version 9c7b6d1, a snapborg configuration like this

~~~ yaml
configs:
  - name: snapper_config_1
    repo: backupuser@borg.example.com
    ...
    retention:
      keep_last: 1
      keep_hourly: 0
      keep_daily: 0
      keep_weekly: 0
      keep_monthly: 6
      keep_yearly: 3
    ...
  - name: snapper_config_2
    repo: backupuser@borg.example.com
    ...
    retention:
      keep_last: 0
      keep_hourly: 0
      keep_daily: 7
      keep_weekly: 0
      keep_monthly: 0
      keep_yearly: 0
    ...
~~~

generates the following two `borg prune` commands for each `snapper backup` run 

~~~ sh
borg prune --list --keep-last 1 --keep-hourly 0 --keep-daily 0 --keep-weekly 0 --keep-montly 6 --keep-yearly 3 backupuser@borg.example.com
borg prune --list --keep-last 10 --keep-hourly 0 --keep-daily 15 --keep-weekly 0 --keep-montly 0 --keep-yearly 0 backupuser@borg.example.com
~~~

which will pretty much wipe out the whole borg repo, only keeping at most 7 archives in total (and that only if the repo started out filled with at least 7 archives that survive the first purge run).

This is probably not what the author of the snapper configuration intended. 

It definitely was not what I intended for my setup, where `snapper_config_1` is a backup of the Linux OS of my laptop (for desaster recovery) with just the last 10 snapshots retained, and `snapper_config_2` has snapshots of my home directory and is meant for long term archival  :smile:

**Comment by rkunze** [comment]
A fix for this is easy: Restrict each prune to only affect the archives created from the corresponding snapper config by adding an appropriate `--glob-archives` option to the `borg prune`.

I have implemented this in https://github.com/rkunze/snapborg/commit/d38d76eb8375d7cead4a440f2dd4c71b03e40b4c and can provide a pull request if you like.

**Comment by enzingerm** [comment]
Thanks for pointing this out, hope no data was lost in the process...
I suppose the borg prune commands do not exactly match the configuration given (don't know where `--keep-last 10` and `--keep-daily 15` are coming from)?

Also I didn't know `user@backuphost` without a path was a valid repo URL, as the borg docs require a path after a colon. So I suppose it's a shortcut for `user@backuphost:.` (cwd).

But anyway, since users might backup different snapper configs to the same borg repo, a fix in the form of a pull request is of course very welcome!

**Comment by rkunze** [comment]
> Thanks for pointing this out, hope no data was lost in the process...

No. I'm currently setting up a new laptop as a replacement for one that died, decided to use the opportunity to replace my homegrown backup scripts by something that is included in the distribution, ended up with the combination of snapper and borgsnap, and did some tests with `--dry-run` to check if it indeed does what I thought it should do.

> I suppose the borg prune commands do not exactly match the configuration given (don't know where --keep-last 10 and --keep-daily 15 are coming from)?

Sorry, my fault. I did the examples by hand and decided halfway through the process to change the example values in the second config. Same for the URL, I just put in some random example value (actually, that one is copied straight out of the comment in https://github.com/enzingerm/snapborg/blob/master/etc/snapborg.yaml).

> But anyway, since users might backup different snapper configs to the same borg repo, a fix in the form of a pull request is of course very welcome!
 
[Done](https://github.com/enzingerm/snapborg/pull/38).

**Comment by enzingerm** [comment]
Thanks for the PR!

Closed by #38
